packages/cli/src/commands/generate/function/function.js

```diff
@@ -3,48 +3,58 @@ import path from 'path'
 import camelcase from 'camelcase'
 import terminalLink from 'terminal-link'
 
-import { getPaths } from 'src/lib'
+import { getPaths, transformTSToJS } from 'src/lib'
 
+import { yargsDefaults } from '../../generate'
 import {
   templateForComponentFile,
   createYargsForComponentGeneration,
 } from '../helpers'
 
-export const files = async ({ name, ...rest }) => {
+export const files = ({ name, ...rest }) => {
+  // Taken from ../component; should be updated to take from the project's configuration
+  const isJavascript = rest.javascript && !rest.typescript
```

Comment:
I'd add `const extension = isJavascript ? '.js' : '.ts'` and then use that on line 22 and 27